acceptance/tests/resource/file/source_attribute.rb

```diff
@@ -164,7 +164,7 @@ class source_test_module {
 
     step "second run should not update file"
     on(agent, puppet('agent', "--test --server #{master}")) do
-      assert_no_match(/content changed/, stdout, "Shouldn't have overwrote any files")
+      assert_no_match(/content changed {(#{Regexp.new(checksums.join('|'))})}/, stdout, "Shouldn't have overwrote any files")
```

Comment:
`checksums` includes all of the different checksum types, but the agent may only update a single file resource, e.g. for `mtime`, so this would make the regexp not match, causing the test to pass when it should fail.